### ISSUE TITLE
Too easy to delete a server :-(

### ISSUE THREAD

**Issue by torfason** [Gamocosm/Gamocosm#40]
On the server starting/stopping update page there is a delete button, and if you push it the server gets deleted without any confirmation or warning. It happened to us and it made us very sad.

I really think it should not be so easy to delete a server. I don't think there should be a delete button on the server start/stop update page, and especially not a button that deletes the server without warning or confirmation.

**Comment by Raekye** [comment]
Sorry about that :S there's a confirmation on the control panel but I forgot to do the same on the updating page. Unfortunately I don't think there's anything you can do about that at this point, but I've made the change. I know it's easy to forget (I'm not good on doing this frequently too), but the first tip is to keep separate backups of your world

I'm trying to think of a better way to automatically keep backups around. I could leave the last snapshot undeleted (so there's always 1 copy around), but when I did something similar originally people didn't like that. But now there's an easy way to delete snapshots (on the main page), I think I might do that

Once again sorry for the trouble >.< , and thanks for reporting this

**Comment by torfason** [comment]
That was a very fast fix! Thanks for responding so quickly, you are doing good work, and of course it is up to us users to test and report, since you are doing this as a volunteer. Besides, it was a relatively new world.

I tested this and did indeed get a confirmation when I tried this with a fresh world. I do wonder, though, if a delete button on the updating page is even needed, especially since there is a very easy way of deleting servers on the main page. It's very easy to mistake the button (on the update page) for some sort of confirmation button (which is what happened to us). So, why not simply remove the delete button on the update page?

Also, even if this did not happen to us, the confirmation popup could well get clicked by mistake (even if the warning is well worded). In some systems, when you choose very dangerous actions (such as deleting the whole server), you are taken to a special page where you need to write a word (such as "DELETE") in a text box before clicking a button to confirm. Perhaps that would be a good solution here (for the main page delete button). I think we will take extra care from now on, but perhaps someone else would be saved some sorrow by doing something like this.

Again, thanks very much for your work, we really do appreciate it!

**Comment by Raekye** [comment]
Ah I see, at least I don't feel as bad - was worried it was a long running world D: way back my friend and I built a huge upside down underground tree in a world, and right after we finished I accidentally deleted the world... Thanks for your words of encouragement :)

I'm thinking all delete servers (except the Digital Ocean stuff - droplets, snapshots, SSH keys) will take you to a separate page, and if you click delete it'll give you the popup confirmation that currently exists. Not as "hard" as typing in a phrase, but I think it's effectively as good and I'm a bit lazy xD

As for why the DO stuff doesn't get a two step process (still has confirmation) - I don't think it's necessary and I'm also lazy xD (I do try to polish Gamocosm as much as possible, but don't do things that are unnecessary - also always open for feedback if someone thinks differently). SSH keys are easy to re-add so they're not a problem. I think it's pretty hard to accidentally delete droplets or snapshots because they're at the bottom of the page and static (opposed to the automatically refreshing server updating page). I agree the delete buttons for servers isn't in a great place, but that's sorta part of a bigger issue which is a UI overhaul. However, my graphic design skills are almost nil and have long been trying to figure out what to do with it

I'll leave this issue open until I implement that (hopefully by the end of this week, busy with school and work), and also for if anyone has any thoughts/alternate ideas

Thanks again for your time! Sorry still about the problem :S it was an oversight on my part
